src/Gravatar.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Baraja\Gravatar;
+
+use Nette\Caching\Cache;
+use Nette\Caching\Storages\FileStorage;
+use Nette\Utils\FileSystem;
+use Nette\Utils\Validators;
+
+
+class Gravatar
+{
+	private ?string $defaultIcon;
+
+	private Cache $cache;
+
+
+	public function __construct(?string $defaultIcon = null, ?string $cacheDir = null)
+	{
+		if ($defaultIcon !== null && Validators::isUrl($defaultIcon) === true) {
+			$this->defaultIcon = $defaultIcon;
+		}
+
+		$cacheDir ??= sys_get_temp_dir() . '/gravatar/' . md5(__DIR__);
+
+		FileSystem::createDir($cacheDir);
+
+		$storage = new FileStorage($cacheDir);
+
+		$this->cache = new Cache($storage, 'User Info');
```

Comment:
Namespace by měl být spíše obecný:

```suggestion
		$this->cache = new Cache($storage, 'gravatar');
```